catalogs/catalog-lakehouse-paimon/src/main/java/org/apache/gravitino/catalog/lakehouse/paimon/GravitinoPaimonTable.java

```diff
@@ -102,6 +121,77 @@ public static Transform[] toGravitinoPartitioning(List<String> partitionKeys) {
     return partitionKeys.stream().map(Transforms::identity).toArray(Transform[]::new);
   }
 
+  private static List<String> getPartitionKeys(Transform[] partitioning) {
+    if (partitioning == null) {
+      return Collections.emptyList();
+    }
+
+    return Arrays.stream(partitioning)
+        .map(
+            partition -> {
+              NamedReference[] references = partition.references();
+              Preconditions.checkArgument(
+                  references.length == 1, "Partitioning column must be single-column, like 'a'.");
+              return references[0].toString();
+            })
+        .collect(Collectors.toList());
+  }
+
+  private List<String> getPrimaryKeysFromIndexes(Index[] indexes) {
+    if (indexes == null || indexes.length == 0) {
+      return Collections.emptyList();
+    }
+
+    Preconditions.checkArgument(
```

Comment:
could you move all the check logic in `checkPaimonPrimaryKeys` and invoke it before create `GravitinoPaimonTable`?

Reply:
this check seems to help better understand `Index primaryKeyIndex = indexes[0];`
WDYT？

Reply:
ok